### DISCUSSION TITLE
Starlite 2.0

### DISCUSSION THREAD

**Discussion by provinzkraut** [litestar-org/litestar#847] [Ideas]
While talking about changing some of our internal workings with @Goldziher, the topic of `2.0` came up, and I wanted to put this out there, as I think it's time we started planning. So, here we go.

Let's start out by collecting and talking about wishes, potential blockers, must-haves and nice-to-haves for `2.0`!

## TBD for 2.0 but can be released in earlier versions
These can already be worked on and released, but should be ready for 2.0

- [Documentation overhaul](https://github.com/orgs/starlite-api/projects/1)
- #768
- #650
- #721
- #707
- Possible renaming (#799)

## Targets 2.0
These are major changes / things that break backwards compatibility hard.

- #846

**Comment by Goldziher** [comment]
One change I want to discuss is to switch __from__ Pydantic __to__ dataclasses as the default fir Starlite. I don't mean remove Pydantic at all, but rather make Pydantic an optional dependency, without which Starlite will default to using the stdlib dataclasses. 

## Why?

While Pydantic is hands down an awesome library, it is relatively heavy and slow. Even with the planned switch to a rust based core for Pydantic 2, it will still remain substantially slower than dataclasses.

For most use cases Pydantic is not required by Starlite. We can in fact get the same functionalities from dataclasses - minus the support for Pydantic specific types and niceties. 

By having the core of starlite itself detached, users will have more freedom to choose which parsing a validation solution to use, if any. And our performance will increase further.

**Reply by provinzkraut** [reply]
I guess we can do this in three steps:

1. Stop implementing new functionality that depends on pydantic and use dataclasses instead
2. Move existing internals away from pydantic
3. Move user-facing code away from pydantic defaults and support generic libraries (#846)

Step 1, and parts of 2 can happen now, part 3 would be a 2.0 thing I guess.
